internal/controller/util/json_util.go

```diff
@@ -232,27 +266,27 @@ func compareValues(val1, val2 interface{}, operator string) (bool, error) {
 	case float64:
 		v2, ok := val2.(float64)
 		if !ok {
-			return false, fmt.Errorf("mismatched types")
+			return false, fmt.Errorf("mismatched types %T and %T", val1, val2)
```

Comment:
This error is not clear - which is the expected value? we can make it clear using:

    fmt.Errorf("type mismatch: expected: %T, actual: %T", expected, actual)

Same for other errors bellow.